_posts/2020-11-22-newpipe-0.20.4.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: post
+title: "NewPipe 0.20.4 released: Important fixes and smaller improvements"
+short: "NewPipe 0.20.4 released"
+date: 2020-11-22 21:00:00 + 1:00
+categories: [pinned, release]
+author: tobigr & Opusforlife
+image: newpipe
+excerpt_separator: <!-- more -->
+---
+
+The last update contained dozens of changes. Although we provided release candidates for testing to a wider audience, some regressions slipped through. This small update addresses critical bugs introduced in the last release, re-adds the option to hide the thumbnail on the lock screen when playing a stream in background mode, and brings a few optimizations "under the hood".
+
+<!-- more -->
+
+## New in this version
+After changing the complete app work flow and navigation in [version 0.20.0]({% post_url 2020-10-04-newpipe-0.20.0 %}), we are still working on improving things and implementing your feedback. Working on the best possible solution which keeps the new UI, but comes with all the features seen in 0.19.x, requires time and sometimes out-of-the-box thinking. We hope to ship some of the old-new buttons and options in the next update. In this version, [@XiangRongLin](https://github.com/XiangRongLin) brought back the option to not show thumbnails on the device's lock screen when playing a stream in the background.
+
+[@okan35](https://github.com/okan35) worked on making NewPipe UI more intuitive and easier to use: when viewing subscriptions and subscription groups, you can now refresh them by simply pulling down form the top of the stream list. 
+
+## Bugs, be gone!
+Well, you might have noticed: we did not catch all bugs in our last release process. We are sorry for that :( Just kidding, we love introducing regressions and new bugs <3  
+Otherwise, we might end up having a real life.
+
+[@Stypox](https://github.com/Stypox) and [@TobiGr](https://github.com/TobiGr) fixed a crash on app startup when there was no internet connection. NewPipe's update check caused this crash. When updating some of its code in the previous release, we forgot to modify the error handling and thus the app crashed fatally when there was a internet connection error. This bug slipped through our release process, because it was specific to NewPipe's release build. Our build-in version check only searches for updates when the APK is a release signed with the release key. Our debug and release candidate builds are signed with different keys.
```

Comment:
```suggestion
[@Stypox](https://github.com/Stypox) and [@TobiGr](https://github.com/TobiGr) fixed a crash on app startup when there was no internet connection. NewPipe's update check caused this crash. When updating some of its code in the previous release, we forgot to modify the error handling and thus the app crashed fatally when there was a internet connection error. This bug slipped through our release process, because it was specific to NewPipe's release build. Our built-in version check only searches for updates when the APK is a release signed with the release key. Our debug and release candidate builds are signed with different keys.
```